src/main/scala/ai/privado/entrypoint/ScanProcessor.scala

```diff
@@ -260,9 +260,17 @@ object ScanProcessor extends CommandProcessor {
     val sinkSkipList = externalConfigAndRules.sinkSkipList ++ internalConfigAndRules.sinkSkipList
     val systemConfig = externalConfigAndRules.systemConfig ++ internalConfigAndRules.systemConfig
     val auditConfig  = externalConfigAndRules.auditConfig ++ internalConfigAndRules.auditConfig
+    val combinedSources = sources
+      .groupBy(_.id)
+      .view
+      .map { case (_, sourceGroup) =>
+        val combinedPattern = sourceGroup.map(_.patterns.mkString("|")).mkString("(", "|", ")")
```

Comment:
You can change this line to `val combinedPatterns = sourceGroup.flatMap(_.patterns)`
and 
`sourceGroup.head.copy(patterns = combinedPatterns)`

Reply:
@Saumya25mis Make a function `mergePatterns(T)` where T can be sources, sinks, collections, policies etc. 
As we need to apply the merging to all the categories